File: Data Structures & Algorithms/frequency-of-the-most-frequent-element/submission-1.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        n = len(nums)
        prefixSum = [0] * (n + 1)
        for i in range(n):
            prefixSum[i + 1] = prefixSum[i] + nums[i]
        
        res = 1

        for i in range(n):
            l, r = 0, i
            while l <= r:
                m = (l + r) // 2
                curSum = prefixSum[i + 1] - prefixSum[m]
                need = (i - m + 1) * nums[i] - curSum
                if need <= k:
                    r = m - 1
                    res = max(res, i - m + 1)
                else:
                    l = m + 1
        
        return res

File: Data Structures & Algorithms/longest-continuous-subarray-with-absolute-diff-less-than-or-equal-to-limit/submission-1.java
class Solution {
    public int longestSubarray(int[] nums, int limit) {
        PriorityQueue<int[]> maxHeap = new PriorityQueue<>(
            (a,b) -> b[0] - a[0]
        );
        PriorityQueue<int[]> minHeap = new PriorityQueue<>(
            (a,b) -> a[0] - b[0]
        );

        int j =0, res = 0;
        for (int i=0; i < nums.length; i++) {
            int v = nums[i];
            maxHeap.offer(new int[]{v, i});
            minHeap.offer(new int[]{v, i});

            while (maxHeap.peek()[0] - minHeap.peek()[0] > limit) {
                ++j;
                while(!maxHeap.isEmpty() && maxHeap.peek()[1] < j){
                    maxHeap.poll();
                }

                while(!minHeap.isEmpty() && minHeap.peek()[1] < j){
                    minHeap.poll();
                }
            }

            res = Math.max(res, i - j + 1);
        }

        return res;
    }
}